modules/runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/model/impl/RegistrationInfoImpl.java

```diff
@@ -342,7 +343,9 @@ protected ComponentInstance createComponentInstance() {
             String msg = "Failed to instantiate component: " + implementation;
             log.error(msg, e);
             msg += " (" + e.toString() + ')';
-            Framework.getRuntime().getMessageHandler().addError(msg);
+            Framework.getRuntime()
+                     .getMessageHandler()
+                     .addMessage(Level.SEVERE, msg, ComponentManagerImpl.class);
```

Comment:
Seems not clear to me why we are using `ComponentManagerImpl.class` here instead of `#getClass()` or `RegistrationInfoImpl.java`.

Reply:
because we need to clear all these in case of hotreload/tests, but i can reference both classes here if you prefer: https://github.com/nuxeo/nuxeo/blob/fix-NXP-29504-runtime-startup-errors/modules/runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/model/impl/ComponentManagerImpl.java#L899

note i'll need to rework this anyway for https://jira.nuxeo.com/browse/NXP-28948, i was trying to merge minimal changes before refactoring the DescriptorRegistry api to handle things more finely (if we want to).

Reply:
source removed -> resolve